packages/ts-moose-lib/src/index.ts

```diff
@@ -4,3 +4,18 @@ export interface Aggregation {
 }
 
 export type Key<T extends string | number | Date> = T;
+
+export enum IngestionFormat {
+  JSON = "JSON",
+  JSON_ARRAY = "JSON_ARRAY",
+}
+
+export type DataModelConfig = Partial<{
```

Comment:
What about making it generic, and the order_by has to be field names of the type parameter?

Reply:
Can you spell it out? I don't see what you are suggesting

Reply:
`order_by_fields?: (keyof T)[];`

And if I mistype the field name:
```
app/datamodels/models.ts:47:23 - error TS2820: Type '"observedTimestam"' is not assignable to type 'keyof Log'. Did you mean '"observedTimestamp"'?

47     order_by_fields: ["observedTimestam"],
```

Then again this is a bit complicated if we want to support nested keys.

Reply:
We can start with that and then iterate